spacesvm/tests/vm/mod.rs

```diff
@@ -0,0 +1,219 @@
+pub mod client;
+
+use avalanche_types::{ids, subnet::rpc::utils};
+use jsonrpc_core::Params;
+use spacesvm::{
+    api::{client::claim_tx, DecodeTxArgs},
+    vm::{self, PUBLIC_API_ENDPOINT},
+};
+use std::io::{Error, ErrorKind};
+use tokio::{
+    sync::{
+        broadcast::{Receiver, Sender},
+        mpsc,
+    },
+    time::{sleep, Duration},
+};
+use tonic::transport::Channel;
+
+#[tokio::test]
+async fn test_api() {
+    // init logger
+    let _ = env_logger::builder()
+        .filter_level(log::LevelFilter::Info)
+        .is_test(true)
+        .try_init();
+
+    // setup stop channel for grpc services.
+    let (stop_ch_tx, stop_ch_rx): (Sender<()>, Receiver<()>) = tokio::sync::broadcast::channel(1);
+    let vm_server = avalanche_types::subnet::rpc::vm::server::Server::new(
+        Box::new(vm::ChainVm::new()),
+        stop_ch_tx,
+    );
+
+    // start Vm service
+    let vm_addr = utils::new_socket_addr();
+    tokio::spawn(async move {
+        avalanche_types::subnet::rpc::plugin::serve_with_address(vm_server, vm_addr, stop_ch_rx)
+            .await
+            .expect("failed to start gRPC server");
+    });
+    log::info!("started subnet vm");
+
+    // wait for server to start
+    sleep(Duration::from_millis(100)).await;
+
+    // create gRPC client for Vm client.
+    let client_conn = Channel::builder(format!("http://{}", vm_addr).parse().unwrap())
+        .connect()
+        .await
+        .unwrap();
+
+    let mut vm_client = crate::vm::client::Client::new(client_conn);
+
+    let mut versioned_dbs: Vec<
+        avalanche_types::subnet::rpc::database::manager::versioned_database::VersionedDatabase,
+    > = Vec::with_capacity(1);
+    versioned_dbs.push(
+        avalanche_types::subnet::rpc::database::manager::versioned_database::VersionedDatabase::new(
+            avalanche_types::subnet::rpc::database::memdb::Database::new(),
+            semver::Version::parse("0.0.7").unwrap(),
```

Comment:
nit: use a constant here as well

Reply:
same as above 007 is just dummy data. bond james bond :)